src/main/java/io/kamax/matrix/client/regular/MatrixHttpClient.java

```diff
@@ -82,6 +82,7 @@ public void login(MatrixPasswordLoginCredentials credentials) {
         LoginResponse response = gson.fromJson(body, LoginResponse.class);
         context.setToken(response.getAccessToken());
         context.setDeviceId(response.getDeviceId());
+        context.setUser(new MatrixID(credentials.getLocalPart(), context.getHs().getDomain()));
```

Comment:
[Not following CS spec](https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#post-matrix-client-r0-login). The Matrix ID is actually part of the answer and should be used.